lib/spack/spack/solver/asp.py

```diff
@@ -409,15 +435,32 @@ def specs(self):
         """List of concretized specs satisfying the initial
         abstract request.
         """
-        # The specs were already computed, return them
-        if self._concrete_specs:
-            return self._concrete_specs
+        if not self._concrete_specs:
+            self._compute_specs_from_answer_set()
+        return self._concrete_specs
 
-        # Assert prerequisite
-        msg = 'cannot compute specs ["satisfiable" is not True ]'
-        assert self.satisfiable, msg
+    @property
+    def unsolved_specs(self):
+        """List of abstract input specs that were not solved."""
+        if not self._unsolved_specs:
```

Comment:
```suggestion
        if self._unsolved_specs is None:
```
That way we don't recompute every time if we already know the list is empty.

Reply:
Fixed in 03e27c2